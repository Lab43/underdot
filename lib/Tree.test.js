const testTree = require('../testing/testTree');



describe('Tree class', () => {
  test('tree matches snapshot', async () => {
    const tree = await testTree();
    expect(tree.tree).toMatchSnapshot();
  })
});
