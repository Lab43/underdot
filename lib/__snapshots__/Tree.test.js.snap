// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree class tree matches snapshot 1`] = `
Object {
  "children": Object {
    "alt": Object {
      "children": Object {
        "alt-page": Object {
          "content": "<p>This is a page using the alt template in a subdirectory.</p>
",
          "dirname": "alt",
          "metadata": Object {
            "template": "alt",
            "title": "Subdirectory Alt Template Page",
          },
        },
        "page": Object {
          "content": "<p>This is a page using the default template in a subdirectory.</p>
",
          "dirname": "alt",
          "metadata": Object {
            "title": "Subdirectory Default Template Page",
          },
        },
      },
    },
    "alt-page": Object {
      "content": "<p>This is a page using the alternative template.</p>
",
      "dirname": "",
      "metadata": Object {
        "template": "alt",
        "title": "Alt Template Page",
      },
    },
    "simple": Object {
      "children": Object {
        "child1": Object {
          "content": "<p>This is a simple child page.</p>
",
          "dirname": "simple",
          "metadata": Object {
            "title": "Simple Child Page 1",
          },
        },
        "child2": Object {
          "content": "<p>This is another simple child page.</p>
",
          "dirname": "simple",
          "metadata": Object {
            "title": "Simple Child Page 2",
          },
        },
      },
      "content": "<p>This is a simple page.</p>
",
      "dirname": "",
      "metadata": Object {
        "title": "Simple Page",
      },
    },
    "sub": Object {
      "children": Object {
        "regular": Object {
          "content": "<p>This is a regular page in a subdirectory.</p>
",
          "dirname": "sub",
          "metadata": Object {
            "title": "Sub Regular Page",
          },
        },
      },
      "content": "<p>This is an index page in a subdirectory.</p>
",
      "dirname": "sub",
      "metadata": Object {
        "title": "Sub Index Page",
      },
    },
  },
  "content": "<p>This is the homepage of Test Site.</p>
",
  "dirname": "",
  "metadata": Object {
    "title": "Home",
  },
}
`;
